File: samples_Python/ex_absdiff.py
import cv2
import numpy as np
x = np.linspace(0, 180, 1000000)
si = np.sin(np.pi*x/180.0)
co = np.cos(np.pi*x/180.0)
t1 = cv2.getTickCount()
for i in range(100):
    r = cv2.absdiff(si, co)
t2 = cv2.getTickCount()
print r
print  (t2-t1)/cv2.getTickFrequency()

t3 = cv2.getTickCount()
for i in range(100):
    r2 = np.abs(si-co)
t4 = cv2.getTickCount()
print r2
print  (t4-t3)/cv2.getTickFrequency()

File: samples_C++/ex_absdiff.cpp
#include <opencv2/opencv.hpp>
void my_absdiff(const cv::Mat &img1, const cv::Mat &img2, cv::Mat &img4){
	img4=cv::Mat::zeros(img1.size(), img1.type());	
	for(int i=0; i<img1.rows; i++){
		for(int j=0; j<img1.cols;j++){
			img4.at<uchar>(i,j) = std::abs(img1.at<uchar>(i,j) - img2.at<uchar>(i,j));
		}
	}
} 

int main(int agrc,  char* argv[]){
	std::string name1 = "../../data/lena.png";
	std::string name2 = "../../data/home.jpg";

	cv::Mat img1=cv::imread(name1, 0);
	cv::Mat img2=cv::imread(name2, 0);

	int64 time0 = cv::getTickCount();
	cv::Mat img3;
	cv::absdiff(img1, img2, img3);
	int64 time1 = cv::getTickCount();
	
	std::cout << (time1-time0)/cv::getTickFrequency() << std::endl;

	int64 time2 = cv::getTickCount();
	
	cv::Mat img4;
	my_absdiff(img1, img2, img4);
	int64 time3 = cv::getTickCount();
	std::cout << (time3-time2)/cv::getTickFrequency() << std::endl;

	cv::imwrite("ex_absdiff.png", img3);
	cv::imwrite("ex_absdiff_loop.png", img4);
}

